src/lib.rs

```diff
@@ -4,8 +4,7 @@ pub use embedded_nal;
 pub use smoltcp;
 
 use embedded_nal::TcpClientStack;
-use smoltcp::dhcp::Dhcpv4Client;
-use smoltcp::socket::AnySocket;
+use smoltcp::socket::{AnySocket, Dhcpv4Event};
```

Comment:
Stylistic nit:

```suggestion
use smoltcp::socket::{AnySocket, Dhcpv4Event, Dhcpv4Socket, SocketHandle};
```

and use those. They come up a couple times.